clang/include/clang/AST/DeclTemplate.h

```diff
@@ -2596,23 +2582,24 @@ class VarTemplateSpecializationDecl : public VarDecl,
   llvm::PointerUnion<VarTemplateDecl *, SpecializedPartialSpecialization *>
   SpecializedTemplate;
 
-  /// Further info for explicit template specialization/instantiation.
-  struct ExplicitSpecializationInfo {
-    /// The type-as-written.
-    TypeSourceInfo *TypeAsWritten = nullptr;
+  struct ExplicitInstantiationInfo {
```

Comment:
This is the 2nd struct here that has the same structure.  Should we instead extract this to a 'details' NS and reuse it for all of them?

Reply:
Yeah, I thought about it but I didn't just want to dump it into the clang namespace a la `QualifierInfo`. I'll address this after the test related changes.